adds new embodiment of Agibot A2D

rebeccazhang0707 opened the pull request:
## Summary
Adds a new embodiment of Agibot

## Detailed description
- What was the reason for the change?
    - Agibot is a widely applied mobile-base bimanual humanoid in China. 
- What has been changed?
    - Adds a new agibot.py

rebeccazhang0707 committed:
adds new embodiment of Agibot A2D

alexmillane commented on isaaclab_arena/embodiments/agibot/agibot.py:
```diff
@@ -0,0 +1,262 @@
+        mimic_arm_mode: MimicArmMode | None = None,
+    ):
+        super().__init__(enable_cameras, initial_pose)
+        self.mimic_arm_mode = mimic_arm_mode or self.default_mimic_arm_mode
```
It seems in this robot the `mimic_arm_mode` is being used for more then mimic. It's defining which arm of the robot is being controlled.

Suggestion to rename the enum `MimicArmMode` -> `ArmMode` and `mimic_arm_mode` -> `arm_mode`.

Does that make sense to you?

rebeccazhang0707 replied:
Absolutely, agree with you @alexmillane. Shall I just rename the MimicArmMode in this MR or create a new one?

alexmillane commented on isaaclab_arena/embodiments/agibot/agibot.py:
```diff
@@ -0,0 +1,262 @@
+                    ),
+                ),
+            ],
+        )
```
This is a bit of a strange use of inheritance. Suggestion to instead create a function which takes in `arm_mode` and constructs a `AgibotSceneCfg` appropriately.

Maybe:

```
@configclass
class AgibotSceneCfg:
    robot = AGIBOT_A2D_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
    arm_mode: ArmMode
    eef_frame: MISSING

   def __post_init__(self):
        if self.arm_mode == ArmMode.LEFT:
               eef = ...
```

Then above in `AgibotEmbodiment` you can do:

```
scene_config = AgibotSceneCfg(arm_mode=ArmMode.LEFT)
```

And we don't need these two classes with inheritance etc.

alexmillane commented on isaaclab_arena/embodiments/agibot/agibot.py:
```diff
@@ -0,0 +1,262 @@
+        joint_names=["right_hand_joint1", "right_.*_Support_Joint"],
+        open_command_expr={"right_hand_joint1": 0.994, "right_.*_Support_Joint": 0.994},
+        close_command_expr={"right_hand_joint1": 0.0, "right_.*_Support_Joint": 0.0},
+    )
```
As above. Suggestion to make a single class which takes an `ArmMode` as input and constructs the config struct appropriately.

alexmillane reviewed (commented):
Thanks for adding this! This is really useful.

I have a couple of comments.

rebeccazhang0707 commented on isaaclab_arena/embodiments/agibot/agibot.py:
```diff
@@ -0,0 +1,262 @@
+        debug_vis=False,
+        target_frames=[
+            FrameTransformerCfg.FrameCfg(
+                prim_path="{ENV_REGEX_NS}/Robot/gripper_center",
```
@alexmillane this `prim_path` is copied from the Agibot usd asset, and I think I will be a little difficult to revise the usd asset already uploaded to AWS S3 bucket. But if you think this is necessary, I can contact the Isaac Sim engineer to revise the asset.

rebeccazhang0707 commented on isaaclab_arena/embodiments/agibot/agibot.py:
```diff
@@ -0,0 +1,262 @@
+                rot_obj_base = PoseUtils.matrix_from_quat(quat_obj_base)
+                object_pose_matrix[art_name] = PoseUtils.make_pose(pos_obj_base, rot_obj_base)
+
+        return object_pose_matrix
```
@alexmillane I modified this func is to align with the observations definition. Since in `AgibotObservationCfg`, our `eef_pos` and `eef_quat `is related to the `robot_base_frame`, which I thought would make more sense if the robot is not located at the env origin or is a mobile robot. So in this MimicEnv, we need to also get_object_poses in `robot_base_frame`. I think reset both ObsCfg and MimicEnv to track `eef_pose` in `env_world_frame `is OK for me, which will simplify this MimicEnv code, just reuse `FrankaMimicEnv`. What is your suggestion?

rebeccazhang0707 committed:
refactors Agibot embodiment file:
- since the property of 'arm_mode' in Agibot is not for building mimic_env_cfg only, we rename MimicArmMode class to ArmMode class

rebeccazhang0707 committed:
Merge branch 'main' into reb/feature/adds_agibot

rebeccazhang0707 committed:
removed get_prompt(), fixed bugs after main merge

rebeccazhang0707 committed:
Merge branch 'main' into reb/feature/adds_agibot
